exonum/src/storage/proof_map_index/key.rs

```diff
@@ -259,52 +263,99 @@ impl PartialEq for DBKey {
 
 impl ::std::fmt::Debug for DBKey {
     fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
-        write!(f, "DBKey(")?;
-        for i in 0..self.len() {
-            write!(
-                f,
-                "{}",
-                match self.get(i) {
-                    ChildKind::Left => '0',
-                    ChildKind::Right => '1',
+        let mut bits = String::with_capacity(KEY_SIZE * 8);
+        for byte in 0..self.data.len() {
+            let chunk = self.data[byte];
+            for bit in (0..8).rev() {
+                let i = (byte * 8 + bit) as u16;
+                match i {
+                    _ if i < self.from => write!(&mut bits, "_")?,
+                    _ if i >= self.to => write!(&mut bits, "_")?,
+                    _ => {
+                        write!(
+                            &mut bits,
+                            "{}",
+                            if (1 << bit) & chunk == 0 { '0' } else { '1' }
+                        )?
+                    }
                 }
-            )?;
+            }
+            write!(&mut bits, "|")?;
         }
-        write!(f, ")")
+
+        f.debug_struct("DBKey")
+            .field("begin", &self.from)
+            .field("end", &self.to)
+            .field("bits", &bits)
+            .finish()
     }
 }
 
+#[test]
+fn test_dbkey_storage_key_leaf() {
+    let key = DBKey::leaf(&[250; 32]);
+    let mut buf = vec![0; DB_KEY_SIZE];
+    key.write(&mut buf);
+    let key2 = DBKey::read(&buf);
+
+    assert_eq!(buf[0], LEAF_KEY_PREFIX);
+    assert_eq!(buf[33], 0);
+    assert_eq!(&buf[1..33], &[250; 32]);
+    assert_eq!(key2, key);
+}
+
+#[test]
+fn test_dbkey_storage_key_branch() {
+    let mut key = DBKey::leaf(&[255; 32]);
```

Comment:
Do you use `leaf` here because you don't have a static factory method for a _branch_?

Reply:
Yes.

Reply:
Won't it make the code clearer (I don't know how frequently it is used) to _add it_?

Reply:
I want to refactor this code in upcoming PR later.